nir/src/main/scala/scala/scalanative/nir/Sig.scala

```diff
@@ -4,7 +4,7 @@ package nir
 import scala.language.implicitConversions
 
 final class Sig(val mangle: String) {
-  final def toProxy: Sig =
+  final lazy val toProxy: Sig =
```

Comment:
After #1536 I was hoping to make `Sig` a value class so that it doesn't require an extra box when stored in `Global.Member`. Adding fields to it would go against this unfortunately. Can you try changing it to value class and checking if it's a win performance wise? Otherwise we can keep it as is in the PR atm.